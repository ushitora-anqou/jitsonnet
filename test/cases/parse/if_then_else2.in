if
  if true then true
  else false
then
  if true then 1.0
  else 1.0
else
  if true then 1.0
  else 1.0
