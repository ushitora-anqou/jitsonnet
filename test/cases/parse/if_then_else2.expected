{ Syntax.v =
  (Syntax.If
     ({ Syntax.v =
        (Syntax.If
           ({ Syntax.v = Syntax.True;
              loc =
              { Syntax.fname = "test/cases/parse/if_then_else2.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 2; column = 5 };
                        endpos = { Syntax.line = 2; column = 9 } })
                }
              },
            { Syntax.v = Syntax.True;
              loc =
              { Syntax.fname = "test/cases/parse/if_then_else2.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 2; column = 15 };
                        endpos = { Syntax.line = 2; column = 19 } })
                }
              },
            (Some { Syntax.v = Syntax.False;
                    loc =
                    { Syntax.fname = "test/cases/parse/if_then_else2.in";
                      ran =
                      (Some { Syntax.startpos =
                              { Syntax.line = 3; column = 7 };
                              endpos = { Syntax.line = 3; column = 12 } })
                      }
                    })));
        loc =
        { Syntax.fname = "test/cases/parse/if_then_else2.in";
          ran =
          (Some { Syntax.startpos = { Syntax.line = 2; column = 2 };
                  endpos = { Syntax.line = 3; column = 12 } })
          }
        },
      { Syntax.v =
        (Syntax.If
           ({ Syntax.v = Syntax.True;
              loc =
              { Syntax.fname = "test/cases/parse/if_then_else2.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 5; column = 5 };
                        endpos = { Syntax.line = 5; column = 9 } })
                }
              },
            { Syntax.v = (Syntax.Number 1.);
              loc =
              { Syntax.fname = "test/cases/parse/if_then_else2.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 5; column = 15 };
                        endpos = { Syntax.line = 5; column = 18 } })
                }
              },
            (Some { Syntax.v = (Syntax.Number 1.);
                    loc =
                    { Syntax.fname = "test/cases/parse/if_then_else2.in";
                      ran =
                      (Some { Syntax.startpos =
                              { Syntax.line = 6; column = 7 };
                              endpos = { Syntax.line = 6; column = 10 } })
                      }
                    })));
        loc =
        { Syntax.fname = "test/cases/parse/if_then_else2.in";
          ran =
          (Some { Syntax.startpos = { Syntax.line = 5; column = 2 };
                  endpos = { Syntax.line = 6; column = 10 } })
          }
        },
      (Some { Syntax.v =
              (Syntax.If
                 ({ Syntax.v = Syntax.True;
                    loc =
                    { Syntax.fname = "test/cases/parse/if_then_else2.in";
                      ran =
                      (Some { Syntax.startpos =
                              { Syntax.line = 8; column = 5 };
                              endpos = { Syntax.line = 8; column = 9 } })
                      }
                    },
                  { Syntax.v = (Syntax.Number 1.);
                    loc =
                    { Syntax.fname = "test/cases/parse/if_then_else2.in";
                      ran =
                      (Some { Syntax.startpos =
                              { Syntax.line = 8; column = 15 };
                              endpos = { Syntax.line = 8; column = 18 } })
                      }
                    },
                  (Some { Syntax.v = (Syntax.Number 1.);
                          loc =
                          { Syntax.fname =
                            "test/cases/parse/if_then_else2.in";
                            ran =
                            (Some { Syntax.startpos =
                                    { Syntax.line = 9; column = 7 };
                                    endpos = { Syntax.line = 9; column = 10 }
                                    })
                            }
                          })));
              loc =
              { Syntax.fname = "test/cases/parse/if_then_else2.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 8; column = 2 };
                        endpos = { Syntax.line = 9; column = 10 } })
                }
              })));
  loc =
  { Syntax.fname = "test/cases/parse/if_then_else2.in";
    ran =
    (Some { Syntax.startpos = { Syntax.line = 1; column = 0 };
            endpos = { Syntax.line = 9; column = 10 } })
    }
  }